### ISSUE TITLE
C_Cpp.files.exclude doesn't work with C_Cpp.caseSensitiveFileSupport "enabled"

### ISSUE THREAD

**Issue by avolden** [microsoft/vscode-cpptools#12672]
### Environment

- OS and Version: W11 22631.4037
- VS Code Version: 1.92.2
- C/C++ Extension Version: 1.22.2

### Bug Summary and Steps to Reproduce

Bug Summary:
When trying to exclude some files from the code analysis by using "C_Cpp.codeAnalysis.exclude", the warnings remain.
I tried different configurations, by writing folder/**, or ${workspaceFolder}/folder/**, but nothing works.
The description of the setting explains it inherits from "files.exclude" and "C_Cpp.files.exclude", but while the "files.exclude" correctly hides the file from the Explorer view and the search, the analysis of the file keeps running.

Steps to reproduce:
1. Create a folder, put a file raising clang-tidy warnings, like :
```cpp
void func()
{
	int* n = 0;
	*n = 0;
}
```
2. Add in the settings.json:
```json
"C_Cpp.codeAnalysis.exclude": {
    "folder/**":true
}
```
3. Notice the clang-tidy warning still displayed, even after a window reload

Expected behavior:
The excluded files should not be checked by the code analysis.

### Configuration and Logs

```shell
-------- Diagnostics - 8/31/2024, 8:47:48 PM
Version: 1.22.2
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "c:/Dev/Sources/C++ tests/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "cStandard": "c17",
    "intelliSenseMode": "windows-clang-x64",
    "cppStandard": "c++17",
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": true,
    "mergeConfigurations": false,
    "compilerPath": "C:\\Dev\\LLVM\\bin\\clang.exe",
    "compilerPathIsExplicit": true,
    "browse": {
        "path": [
            "c:/Dev/Sources/C++ tests/**",
            "${workspaceFolder}"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Modified Settings:
{
    "C_Cpp.default.compilerPath": "C:\\Dev\\LLVM\\bin\\clang.exe",
    "C_Cpp.default.cppStandard": "c++17",
    "C_Cpp.exclusionPolicy": "checkFilesAndFolders",
    "C_Cpp.files.exclude": {
        "**/.vscode": true,
        "**/.vs": true,
        "folder/test.c": true
    },
    "C_Cpp.formatting": "clangFormat",
    "C_Cpp.doxygen.generateOnType": false,
    "C_Cpp.codeAnalysis.exclude": {
        "${workspaceFolder}/folder/**": true
    },
    "C_Cpp.codeAnalysis.clangTidy.enabled": true,
    "C_Cpp.codeAnalysis.clangTidy.path": "C:\\Dev\\LLVM\\bin\\clang-tidy.exe",
    "C_Cpp.intelliSenseMemoryLimit": 16384,
    "C_Cpp.loggingLevel": "Debug",
    "C_Cpp.experimentalFeatures": "enabled",
    "C_Cpp.caseSensitiveFileSupport": "enabled"
}
Additional Tracked Settings:
{
    "editorTabSize": 4,
    "editorInsertSpaces": false,
    "editorAutoClosingBrackets": "languageDefined",
    "filesEncoding": "utf8",
    "filesAssociations": {
        "*.sln": "xml",
        "log": "log",
        "iostream": "cpp"
    },
    "filesExclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/CVS": true,
        "**/.DS_Store": true,
        "**/Thumbs.db": true,
        "folder/**": true
    },
    "filesAutoSaveAfterDelay": false,
    "editorInlayHintsEnabled": true,
    "editorParameterHintsEnabled": true,
    "searchExclude": {
        "**/node_modules": true,
        "**/bower_components": true,
        "**/*.code-search": true
    },
    "workbenchSettingsEditor": "ui"
}
cpptools version (native): 1.22.2.0
Translation Unit Mappings:
[ C:\Dev\Sources\C++ tests\folder\test.c - source TU]:
Translation Unit Configurations:
[ C:\Dev\Sources\C++ tests\folder\test.c ]
    Process ID: 30248
    Memory Usage: 59 MB
    Compiler Path: C:\Dev\LLVM\bin\clang.exe
    Includes:
        C:\Dev\Sources\C++ tests
        C:\Dev\Sources\C++ tests\folder
        C:\Dev\Sources\C++ tests\src
    System Includes:
        C:\Dev\LLVM\lib\clang\18\include
        C:\Dev\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include
        C:\Dev\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\atlmfc\include
        C:\Windows Kits\10\Include\10.0.22621.0\ucrt
        C:\Windows Kits\10\Include\10.0.22621.0\shared
        C:\Windows Kits\10\Include\10.0.22621.0\um
        C:\Windows Kits\10\Include\10.0.22621.0\winrt
        C:\Windows Kits\10\Include\10.0.22621.0\cppwinrt
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: c17
    IntelliSense Mode: windows-clang-x64
    Other Flags:
        --clang
        --clang_version=180108
        --ms_compatibility
Total Memory Usage: 59 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 5398
```

**Comment by BluTree** [comment]
The folder exclusion doesn't work, I still get the analyser warnings with `"folder":true`.
The exclusion policy was already set in `checkFilesAndFolders`, as seen in the logs sent. I tried different syntax for the file exclusion, with inherited `C_Cpp.files.exclude` and `files.exclude` but this doesn't seem to work either.

**Comment by sean-mcmanus** [comment]
Hmm...I still don't repro the issue. I'll see if anyone else on our team can repro it.

**Comment by Yingzi1234** [comment]
Hi @BluTree @sean-mcmanus, we have been verified this issue on latest c/c++ version v1.22.2(pre-release), it is not reproduced. You can get the repro steps below, please let us know if it is wrong. Thank you!

**ENV:**
C/C++: 1.22.2(pre-release)
VSCode: 1.92.2
CMake tools: v1.20.3(pre-release)

**Repro steps:**
1. Open this folder with VSCode[test.zip](https://github.com/user-attachments/files/16862254/test.zip)
2. Click F1 to run command "cmake: command"
3. Check if error occurs in output window (c/c++ and cmake/build)
4. Check for errors in the Example.cpp file
![image](https://github.com/user-attachments/assets/a181ac10-32d6-4c21-9016-2401b9610d0c)

**Comment by BluTree** [comment]
When opening the folder from the test, and initializing the CMakeLists with the same extensions versions, and the clang-tidy warning is still present.

After investigating, the problem occurs when `C_Cpp.caseSensitiveFileSupport` is enabled.
With `"C_Cpp.caseSensitiveFileSupport": "enabled"`:
![image](https://github.com/user-attachments/assets/ace0d9e4-7e9e-4966-9d84-d117eb763287)

With `"C_Cpp.caseSensitiveFileSupport": "default"`:
![image](https://github.com/user-attachments/assets/80cac94b-3a60-4343-84f1-f66fd94364af)

**Comment by Yingzi1234** [comment]
Hi @BluTree Thanks for adding, as per your information I can reproduce this issue after enable “`C_Cpp.caseSensitiveFileSupport"C_Cpp.caseSensitiveFileSupport": "enabled`"” option.

@sean-mcmanus Based on the user's information we can reproduce the problem, we can reproduce the problem according to the reproduction steps above. Please let us know if you have any concerns!

![image](https://github.com/user-attachments/assets/e7ebe256-fd4a-41ef-8b44-0123dc5cb950)

**Comment by sean-mcmanus** [comment]
@Yingzi1234 Ah, yeah, I didn't see the caseSensitiveFileSupport setting enabled. We've seen multiple bugs related to that setting recently with 1.22.x.

**Comment by sean-mcmanus** [comment]
@BluTree Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.22.4 . But there might be other bugs in the case-sensitive scenario on Windows: https://github.com/microsoft/vscode-cpptools/issues/12732

**Comment by Yingzi1234** [comment]
Hi @BluTree @sean-mcmanus , this issue has been fixed on latest C/C+ version v1.22.4(pre-release), you can get the details below. Thank you!

**ENV:**
1. VSCode version: v1.93.1
2. C/C++ version: v1.22.4(pre-release)

![image](https://github.com/user-attachments/assets/c389ad7b-3c4c-4ab3-88f4-0378468b71dd)
